addons/mail/static/tests/helpers/test_utils.js

```diff
@@ -164,6 +164,21 @@ function _useDiscuss(callbacks, { afterNextRender }) {
 // Public: rendering timers
 //------------------------------------------------------------------------------
 
+/**
+ * @param {string} selector
+ * @param {string} content
+ */
+async function insertText(selector, content) {
+    await afterNextRender(() => {
+        document.querySelector(selector).focus();
+        for (const char of content) {
+            document.execCommand('insertText', false, char);
+            document.querySelector(selector).dispatchEvent(new window.KeyboardEvent('keydown', { key: char }));
+            document.querySelector(selector).dispatchEvent(new window.KeyboardEvent('keyup', { key: char }));
```

Comment:
I wonder if it works with the dispatches only once after the loop (probably not, but worth a try).

Reply:
It doesn't, unfortunately